pkg/kubelet/kubelet_node_status.go

```diff
@@ -404,18 +407,29 @@ func (kl *Kubelet) tryUpdateNodeStatus(tryNumber int) error {
 		return fmt.Errorf("nil %q node object", kl.nodeName)
 	}
 
+	nodeHasChanged := false
 	if node.Spec.PodCIDR != "" {
+		nodeHasChanged = true
```

Comment:
if updatePodCIDR would fail, theoretically node could not change.
But I'm assuming in that case, it doesn't cost us much to report status anyway, so better to be on the safe side.

Maybe worth adding a comment?